### ISSUE TITLE
Functions called in cuvis.hpp, undefined / undeclared in cuvis.h (nor elsewhere in the scope)

### ISSUE THREAD

**Issue by mhmdjouni** [cubert-hyperspectral/cuvis.cpp#28]
I tried running it first in the C version of the examples, no issues. In the C++ version though it gives some errors.

Steps to reproduce:
* Clone the repository cuvis.cpp.examples somewhere on machine (mine is Linux Ubuntu 22.04)
* Inside the cuvis.cpp.examples directory, run "git submodule update --init"
* In terminal, run
  * mkdir build
  * cd build/
  * cmake ..

A bunch of errors of the same type occur (see below), for which it seems the following functions are reported undeclared in the scope:
* `cuvis_acq_cont_get_pixel_format(*_acqCont, id, format)`
* `cuvis_acq_cont_set_pixel_format(*_acqCont, id, format.c_str())`
* `cuvis_acq_cont_get_available_pixel_format_count(*_acqCont, id, &count)`
* `cuvis_acq_cont_get_pixel_format(*_acqCont, id, format)`

In particular, here is the full report:

```
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
statusDoxygen is needed to build the documentation.
Failed to compile and run get_version_executable
CMake Error at /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/FindCuvisCpp.cmake:91 (message):
  Change Dir:
  '/home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/01_loadMeasurement_cpp/build/try_compile/CMakeFiles/CMakeTmp'




  Run Build Command(s): /usr/local/bin/cmake -E env VERBOSE=1 /usr/bin/gmake
  -f Makefile cmTC_41651/fast

  /usr/bin/gmake -f CMakeFiles/cmTC_41651.dir/build.make
  CMakeFiles/cmTC_41651.dir/build

  gmake[1]: Entering directory
  '/home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/01_loadMeasurement_cpp/build/try_compile/CMakeFiles/CMakeTmp'


  Building CXX object CMakeFiles/cmTC_41651.dir/get_version.cpp.o

  /usr/bin/c++ -isystem
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface
  -isystem
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/auxiliary/include
  -pthread -o CMakeFiles/cmTC_41651.dir/get_version.cpp.o -c
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/helper/get_version.cpp


  In file included from
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/helper/get_version.cpp:2:



  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:
  In member function ‘std::string
  cuvis::AcquisitionContext::get_pixel_format(cuvis::int_t) const’:


  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:2368:9:
  error: ‘cuvis_acq_cont_get_pixel_format’ was not declared in this
  scope; did you mean ‘cuvis_acq_cont_get_state’?

   2368 |     chk(cuvis_acq_cont_get_pixel_format(*_acqCont, id, format));
        |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        |         cuvis_acq_cont_get_state

  
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:
  In member function ‘void
  cuvis::AcquisitionContext::set_pixel_format(cuvis::int_t, std::string)’:


  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:2374:9:
  error: ‘cuvis_acq_cont_set_pixel_format’ was not declared in this scope

   2374 |     chk(cuvis_acq_cont_set_pixel_format(*_acqCont, id, format.c_str())); }
        |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:
  In member function ‘std::vector<std::__cxx11::basic_string<char> >
  cuvis::AcquisitionContext::get_available_pixel_formats(cuvis::int_t)
  const’:


  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:2379:9:
  error: ‘cuvis_acq_cont_get_available_pixel_format_count’ was not
  declared in this scope; did you mean
  ‘cuvis_comp_available_pixel_format_get’?

   2379 |     chk(cuvis_acq_cont_get_available_pixel_format_count(*_acqCont, id, &count));
        |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        |         cuvis_comp_available_pixel_format_get

  
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/interface/cuvis.hpp:2384:11:
  error: ‘cuvis_acq_cont_get_pixel_format’ was not declared in this
  scope; did you mean ‘cuvis_acq_cont_get_state’?

   2384 |       chk(cuvis_acq_cont_get_pixel_format(*_acqCont, id, format));
        |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        |           cuvis_acq_cont_get_state

  gmake[1]: *** [CMakeFiles/cmTC_41651.dir/build.make:78:
  CMakeFiles/cmTC_41651.dir/get_version.cpp.o] Error 1

  gmake[1]: Leaving directory
  '/home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/01_loadMeasurement_cpp/build/try_compile/CMakeFiles/CMakeTmp'


  gmake: *** [Makefile:127: cmTC_41651/fast] Error 2



Call Stack (most recent call first):
  /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/cuvis.cpp/FindCuvisCpp.cmake:98 (get_library_version)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!

**Comment by mhmdjouni** [comment]
A naïve way to bypass these errors is to comment out the lines 2368, 2374, 2379, and 2384 of `cuvis.hpp`, which then allows to build the binary file using the `make` command. However:
* Running the command `./01_loadMeasurement_cpp ./../../../../sample_data/set_examples/settings/ ./../../../../sample_data/set_examples/set0_single/single.cu3s` gives the output:
```
Example 01 load measurement cpp 
User Settings Dir: ./../../../../sample_data/set_examples/settings/
sessionfile (.cu3s): ./../../../../sample_data/set_examples/set0_single/single.cu3s
loading user settings...
loading session... 
No CUDA device found.
loading measurement... 
Data 10000_copy t=30 ms mode=2 
  Flags
  - Flag_DataUsesPoorReference (146492)
01_loadMeasurement_cpp: /home/jounim/Work/codebase_projects/cuvis.sdk/examples/cuvis.cpp.examples/01_loadMeasurement_cpp/main.cpp:51: int main(int, char**): Assertion `mesu.get_meta()->processing_mode == Cube_Raw && "This example requires raw mode"' failed.
Aborted (core dumped)
```
* Running the output `./01_loadMeasurement_cpp ./../../../../sample_data/set_examples/settings/ ./../../../../sample_data/set_examples/set0_single/single_raw.cu3s` gives the output:
```
Example 01 load measurement cpp 
User Settings Dir: ./../../../../sample_data/set_examples/settings/
sessionfile (.cu3s): ./../../../../sample_data/set_examples/set0_single/single_raw.cu3s
loading user settings...
loading session... 
No CUDA device found.
loading measurement... 
Data 10000_copy t=30 ms mode=0 
lambda [nm]; raw counts [au] 
350; 0
354; 0
358; 0
362; 0
366; 0
370; 0
374; 0
378; 0
382; 5
386; 11
390; 19
394; 34
398; 49
402; 68
406; 86
410; 106
414; 128
418; 150
422; 172
426; 195
430; 217
434; 241
438; 264
442; 293
446; 327
450; 359
454; 377
458; 395
462; 422
466; 453
470; 481
474; 507
478; 532
482; 565
486; 598
490; 632
494; 668
498; 704
502; 736
506; 768
510; 788
514; 797
518; 811
522; 855
526; 899
530; 916
534; 899
538; 881
542; 957
546; 1042
550; 1039
554; 1022
558; 1010
562; 1009
566; 1009
570; 1014
574; 1020
578; 1010
582; 988
586; 966
590; 989
594; 1033
598; 1071
602; 1047
606; 1022
610; 1017
614; 1023
618; 1029
622; 1100
626; 1177
630; 1227
634; 1206
638; 1188
642; 1177
646; 1166
650; 1172
654; 1192
658; 1211
662; 1248
666; 1287
670; 1327
674; 1285
678; 1237
682; 1192
686; 1148
690; 1118
694; 1168
698; 1208
702; 1154
706; 1101
710; 1050
714; 1005
718; 968
722; 1026
726; 1083
730; 1138
734; 1178
738; 1232
742; 1332
746; 1433
750; 1495
754; 1532
758; 1571
762; 1610
766; 1656
770; 1725
774; 1794
778; 1808
782; 1726
786; 1643
790; 1542
794; 1440
798; 1356
802; 1353
806; 1349
810; 1310
814; 1241
818; 1187
822; 1238
826; 1288
830; 1285
834; 1243
838; 1201
842; 1179
846; 1170
850; 1130
854; 1060
858; 989
862; 923
866; 860
870; 808
874; 808
878; 808
882; 808
886; 807
890; 792
894; 776
898; 733
902; 670
906; 607
910; 554
914; 522
918; 495
922; 481
926; 467
930; 427
934; 386
938; 356
942; 327
946; 298
950; 273
954; 255
958; 235
962; 213
966; 194
970; 183
974; 173
978; 162
982; 142
986; 127
990; 117
994; 108
998; 105
1002; 105
Cuvis is shutting down...
finished. 
Segmentation fault (core dumped)
```

**Comment by birkholz-cubert** [comment]
Hello,

so this functions still had the wrong names in the 3.3.0beta1 version. This is now fixed in the 3.3.0rc1 version.
The functions are new called:
- cuvis_comp_pixel_format_get
- ...

It seems like the submodule for the cuvis.cpp.examples repository was not updated.
The submodule should now point to the correct repository
